Third-party-libraries.md

```diff
@@ -0,0 +1,255 @@
+
+Backend libraries that are not installed using pip also have their versions specified in `install_third_party_libs.py`, but they use custom code for each library. You should read the code to find the section for your library.
+
+### Upgrade frontend libraries
+
+#### Upgrade frontend libraries installed from npm
+
+You can update all frontend libraries that we install from npm as follows. Note that `<yarn version>` specifies the currently-installed version of yarn (there should only be one version in `oppia_tools`). Also, all commands should be run from the repository root.
+
+1. Run `../oppia_tools/yarn-<yarn version>/bin/yarn upgrade` to upgrade libraries that should not have any breaking changes.
+2. Run `../oppia_tools/yarn-<yarn version>/bin/yarn outdated` to show outdated libraries whose newer versions might have breaking changes.
+3. Check for breaking changes in the outdated libraries from the previous step. You should:
+
+   * Check the library's changelog for breaking changes.
+   * Test that after you install the upgraded version (see below), everything works correctly.
+
+4. Manually update the versions in `package.json` for all packages you decide to upgrade.
+5. Run `../oppia_tools/yarn-<yarn version>/bin/yarn install`
+
+#### Upgrade frontend libraries not installed from npm
+
+Other frontend libraries are specified in `manifest.json`. You can upgrade these libraries as follows:
+
+1. Check for breaking changes in the libraries you want to upgrade. You should:
+
+   * Check the library's changelog for breaking changes.
+   * Test that after you install the upgraded version (see below), everything works correctly.
+
+2. Change the version in `manifest.json` to the new version you want to install.
+
+Note that we don't have an automatic way to detect outdated libraries in `manifest.json`, so to upgrade all such libraries, you have to check each version manually.
+
+## Add dependencies
+
+Note that all dependencies must be compatible with our Apache 2 license. All additions must also be approved by @vojtechjelinek.
+
+### Add a backend library
+
+Note that if a library is needed both for development and in production, then it should be added according to both of the following sets of instructions.
+
+#### Add a backend, production library from pip
+
+Here's how to add a backend, production library that can be installed from pip:
+
+1. Add the library to `requirements.in` in the form `<package-name>==<version>`.
+2. Generate `requirements.txt` from `requirements.in` by running `scripts/regenerate_requirements.py`.
+
+#### Add a backend, development library from pip
+
+To add a backend, development library that can be installed from pip, first ask whether the library is needed before running or importing any script. If the answer is yes, add it to the `PREREQUISITES` constant in `scripts/install_third_party_libs.py`. Otherwise, add it to the `local_pip_dependencies` variable in `scripts/install_third_party_libs.py`.
```

Comment:
ask --> figure out

How does one figure this out? Maybe elaborate on that a bit.

Reply:
done